main.py

```diff
@@ -0,0 +1,1605 @@
+
+    # remember subscribed topics so when we select a device, we automatically subscribe to them again
+    # save to .txt file 
+    # write a new function that will read from that file every time we select a device
+    # to check if there are any topics we were watching previously
+    
+    '''
+
+def tab2_layout(tab2):
+    global tab2_listBox
+    global tab2_msg_input
+    global tab2_topic_input
+    global tab2_messages_list
+
+    #tab2 layout
+    tab2.columnconfigure(tuple(range(7)), weight=1) #eight columns
+    tab2.rowconfigure(tuple(range(3)), weight=1)    #three rows
+
+    tab2_select_label = Label(tab2, text='Select a topic to send a message to: ')
+    tab2_select_label.config(font=(myFont, 13), background=myGrey)
+    tab2_select_label.grid(column=0, row=0, columnspan=4, padx=(0,5), pady=(15,0), ipady=5, sticky=N+S)
+
+    tab2_listbox_border = Frame(tab2, background='white')
+    tab2_listbox_border.grid(column=0, row=1, columnspan=4, rowspan=2, padx=(40,5), pady=(5,15),
+                        sticky=W+E+N+S)
+
+    tab2_listBox = Listbox(tab2_listbox_border, borderwidth=0, highlightthickness=0, background='white')
+    tab2_scroll = Scrollbar(tab2_listbox_border, orient=VERTICAL, command=tab1_listBox.yview)
+    tab2_listBox.config(font=(myFont, 12), yscrollcommand=tab2_scroll.set, relief=RAISED, 
+                        selectbackground='#737c7d')
+    
+    tab2_scroll.pack(side=RIGHT, fill=Y)
+    tab2_listBox.pack(padx=20, pady=20, fill=BOTH, expand=TRUE)
+
+    #create list of "Publish" topics to insert into Listbox
+    pub_topic_list = [
+        '{device_id}/shadow/get/accepted',
+        '$aws/things/{device_id}/shadow/get/rejected',
+        '$aws/things/{device_id}/shadow/update/delta',
+        'mqtt_topic_prefix/m/d/{device_id}/c2d',
+        'mqtt_topic_prefix/{device_id}/jobs/rcv',
+        'm/#',
+        'a/connections'
+    ]
+    tab2_update_listBox(pub_topic_list)  #add topics to our list
+    tab2_listBox.bind("<<ListboxSelect>>", tab2_fillOut)  #create a binding on the listbox onclick
+
+    tab2_help = Label(tab2, text="[ Help ]", background='#80AABA', foreground='white')
+    tab2_help.config(font=(myFont, 12), borderwidth=1, relief='raised', anchor=CENTER, justify=LEFT)
+    tab2_help.grid(column=0, row=3, padx=(40,0), pady=(5,15), ipadx=2, sticky=N+S)
+    ToolTip(tab2_help, msg='Click on a topic and a message\nabove to auto-fill or manually\ntype into the textbox.',
+            background='#AEDAEB', font=(myFont, 12), follow=True)
+
+    tab2_topic_input = Entry(tab2)  #this is for the topic
+    tab2_topic_input.config(font=(myFont, 12))
+    tab2_topic_input.insert(0, 'Enter a topic...')  #insert shadow text
+    tab2_topic_input.grid(column=1, row=3, columnspan=3, padx=5, pady=(5,15), sticky=W+E)
+
+    #remove/show shadow text when in or out of focus
+    tab2_topic_input.bind('<FocusIn>', tab2_on_entry_click_left)
+    tab2_topic_input.bind('<FocusOut>', tab2_entry_focus_out_left)
```

Comment:
Using lambdas does come with some downsides, though. This assumes that nothing other than direct user interaction can cause the shadow text to disappear. That assumption holds true for most of your codebase, but in some sections you have bulk clear buttons which undo this assumption

Therefore, if you were to restructure this code, I'd actually recommend properly subclassing the Input class rather than using closures. But this is still a useful design pattern to be aware of